lms/static/scripts/frontend_apps/components/dashboard/DashboardActivityFilters.tsx

```diff
@@ -145,9 +143,21 @@ function SegmentsMultiSelect({ segments }: { segments: SegmentsSelection }) {
       value={segments.selectedIds}
       onChange={newSegmentIds => segments.onChange(newSegmentIds)}
       disabled={segments.entries.length === 0}
+      buttonClasses={classnames({
+        // Reduce button vertical padding to compensate for the count badge
+        // padding
+        'py-1':
+          !isNoneType &&
+          segments.selectedIds.length === 0 &&
+          segments.entries.length > 0,
+      })}
       buttonContent={
-        segments.selectedIds.length === 0 ? (
-          <>{allSegmentsText}</>
+        isNoneType ? (
+          'No sections/groups'
+        ) : segments.selectedIds.length === 0 ? (
+          <ContentWithBadge count={segments.entries.length}>
```

Comment:
It looks like we currently don't paginate the list of segments. Is there any chance that we will need to in future?

Reply:
I guess there is, but we don't get this information from a dedicated endpoint like the rest of the filters (I reckon it was tricky from the BE to do it like that), so if it is paginated at some point, we will have to change the FE logic anyway.